promql/parse.go

```diff
@@ -1074,13 +1074,19 @@ func (p *parser) checkType(node Node) (typ model.ValueType) {
 
 	case *Call:
 		nargs := len(n.Func.ArgTypes)
-		if na := nargs - n.Func.OptionalArgs; na > len(n.Args) {
+		if na := nargs - n.Func.Variadic; n.Func.Variadic >= 0 && na > len(n.Args) {
```

Comment:
A comment on what we are checking before each of the conditions here would be helpful in the future.